tests/e2e/specs/setup-mc/step-4-complete-campaign.test.js

```diff
@@ -0,0 +1,569 @@
+					const lowBudgetTip = setupBudgetPage.getLowerBudgetTip();
+					await expect( lowBudgetTip ).toBeVisible();
+				} );
+
+				test( 'should not see the low budget tip when the buget is set higher than the recommended value', async () => {
+					await setupBudgetPage.fillBudget( '99999' );
+					const lowBudgetTip = setupBudgetPage.getLowerBudgetTip();
+					await expect( lowBudgetTip ).not.toBeVisible();
+				} );
+			} );
+
+			test.describe( 'Set up billing', () => {
+				let newPage;
+
+				test( 'should see set up billing button is enabled', async () => {
+					const setUpBillingButton =
+						setupBudgetPage.getSetUpBillingButton();
+					await expect( setUpBillingButton ).toBeEnabled();
+				} );
+
+				test( 'should see the correct set up billing link', async () => {
+					const setUpBillingLink =
+						setupBudgetPage.getSetUpBillingLink();
+					await expect( setUpBillingLink ).toHaveAttribute(
+						'href',
+						'https://support.google.com/google-ads/answer/2375375'
+					);
+				} );
+
+				test( 'should open a popup when clicking set up billing button', async () => {
+					const popupPromise = page.waitForEvent( 'popup' );
+					await setupBudgetPage.clickSetUpBillingButton();
+					const popup = await popupPromise;
+					await popup.waitForLoadState();
+					const popupTitle = await popup.title();
+					const popupURL = await popup.url();
+					await expect( popupTitle ).toBe(
+						'Add a new payment method in Google Ads - Google Ads Help'
+					);
+					await expect( popupURL ).toBe(
+						'https://support.google.com/google-ads/answer/2375375'
+					);
+					await popup.close();
+				} );
+
+				test( 'should open a new page when clicking set up billing link', async () => {
+					const newPagePromise = page.waitForEvent( 'popup' );
+					await setupBudgetPage.clickSetUpBillingLink();
+					newPage = await newPagePromise;
+					await newPage.waitForLoadState();
+					const newPageTitle = await newPage.title();
+					const newPageURL = await newPage.url();
+					await expect( newPageTitle ).toBe(
+						'Add a new payment method in Google Ads - Google Ads Help'
+					);
+					await expect( newPageURL ).toBe(
+						'https://support.google.com/google-ads/answer/2375375'
+					);
+				} );
+
+				test( 'should see billing has been set up successfully when billing status API returns approved', async () => {
+					await setupBudgetPage.fulfillBillingStatusRequest( {
+						status: 'approved',
+					} );
+
+					const responsePromise = page.waitForResponse(
+						/\/gla\/ads\/billing-status/
+					);
+
+					await newPage.close();
+
+					const billingSetupSection =
+						setupBudgetPage.getBillingSetupSection();
+					await billingSetupSection.focus();
+
+					await responsePromise;
+
+					const billingSetupSuccessSection =
+						setupBudgetPage.getBillingSetupSuccessSection();
+					await expect( billingSetupSuccessSection ).toContainText(
+						'Billing method for Google Ads added successfully'
+					);
```

Comment:
This test takes approximately 20 seconds to run locally, I think it is because it waits for the `billing-status` request to trigger. Additionally, when I manually debug the test, it gets stuck because the `billingSetupSection.focus()` is executed too late, and it displays the `Billing method for Google Ads added successfully` message first. It might be easier to reload the page and test whether it correctly displays the "Billing method for Google Ads added successfully" message.